app/controllers/api/automate_domains_controller.rb

```diff
@@ -1,5 +1,30 @@
 module Api
   class AutomateDomainsController < BaseController
+
+    REQUIRED_FIELDS = %w(git_url ref_type ref_name)
+
+    def create_resource(type, _id, data)
```

Comment:
My only concern here is if there are other ways to creates domains.  That is, I understand that the UI doesn't use the REST API now, but if it did in the future, then we would need a more pure "create" method, and it would be "taken" by this method.  If this PR doesn't preclude something like that then, I think it's fine.

Here is my writeup from ages ago from how I thought this should be designed.  https://github.com/ManageIQ/manageiq/pull/9517#issuecomment-232982860 .  I *think* the way we have it here is ok, but it might be better to have an alternate action like "import_from_git" or something, just so we don't clobber the actual "create" action.

Reply:
I read the associated issue to this: https://github.com/ManageIQ/manageiq-api/issues/354, and there is also a request for importing yaml or zip through the API as well, which also goes to my question about whether this should be "create" or something else.

Reply:
@Fryguy Thank you for the help. I believe I've addressed the suggestions/issues you raised.